ryzen-monitor-ng: excise (obsolete) binaries from upstream source

PhDyellow opened the pull request:
## Description of changes
Fix for ryzen_monitor_ng, binaries and .o files were committed to repo. `makeTargets` is not used by the default builder, so `make` was not building the binary. The installed binary was the one committed to the upstream repo, which is out of date.

The best solution seems to be `lib.cleanSource`, since the problem is .o files committed to the repo that should not be used by NixOS.

@Cryolitia 

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Cryolitia commented:
nixpkgs-review fault:

```
error:
       … while querying the derivation named 'ryzen-monitor-ng-2.0.5-unstable-2023-11-05'

       … while calling the 'getAttr' builtin

         at /builtin/derivation.nix:19:19: (source not available)

       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'ryzen-monitor-ng-2.0.5-unstable-2023-11-05'
         whose name attribute is located at /home/cryolitia/.cache/nixpkgs-review/pr-310507/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:331:7

       … while evaluating attribute 'src' of derivation 'ryzen-monitor-ng-2.0.5-unstable-2023-11-05'

         at /home/cryolitia/.cache/nixpkgs-review/pr-310507/nixpkgs/pkgs/by-name/ry/ryzen-monitor-ng/package.nix:10:3:

            9|   # adds support for Matisse AMD CPUs.
           10|   src = lib.cleanSource (fetchFromGitHub {
             |   ^
           11|     owner = "plasmin";

       error: cannot build '/nix/store/77y3gs2h9fm1wv4py3790my4bx7ba90x-source.drv^out' during evaluation because the option 'allow-import-from-derivation' is disabled
https://github.com/NixOS/nixpkgs/pull/310507 failed to build: Failed to list packages: nix-env failed with exit code 1
```

seems the same problem that ofborg fault......

Aleksanaa commented:
`fetchFromGitHub` creates a FOD (Fixed Output Derivation), and `lib.cleanSource` works by reading directory in Nix eval process (`builtins.readDir`), not in build process, so you end up making eval process relying on the artifact in store, as so called IFD (Input From Derivation). Currently IFD is not allowed in Nixpkgs due to backward compatibility and Hydra's heavy workload.

The most appropriate approach here is to delete these binaries in `postPatch` or `preBuild`. If you are writing flake in the upstream source code repository, using `lib.cleanSource` can indeed solve the problem.

PhDyellow commented:
> `fetchFromGitHub` creates a FOD (Fixed Output Derivation), and `lib.cleanSource` works by reading directory in Nix eval process (`builtins.readDir`), not in build process, so you end up making eval process relying on the artifact in store, as so called IFD (Input From Derivation). Currently IFD is not allowed in Nixpkgs due to backward compatibility and Hydra's heavy workload.
> 
> The most appropriate approach here is to delete these binaries in `postPatch` or `preBuild`. If you are writing flake in the upstream source code repository, using `lib.cleanSource` can indeed solve the problem.

Thanks for that, I've learned something. I think I know what to do from here.

Commit:
ryzen-monitor-ng: Fix build

makeTargets is not used by the default builder, and was not
calling the necessary make targets.

Added `make clean` to `preBuild` to ensure old files are removed
before building.

PhDyellow commented:
I won't close it, there is a bug that needs fixing. `'lib.cleanSource` just wasn't the way to solve it. 

`lib.cleanSource` is still useful to know about, it is elegant and it did work on my machine.

Aleksanaa commented on pkgs/by-name/ry/ryzen-monitor-ng/package.nix:
```diff
@@ -14,13 +14,13 @@ stdenv.mkDerivation {
 
-  ## Remove binaries committed into upstream repo
+  ## Remove binaries and object files
+  ## committed into upstream repo
```
You'd better putting "committed" in the last line to avoid ambiguity, Otherwise, others will think you are saying "I have submitted this fix to the upstream repository"

PhDyellow replied:
Thank you, I have amended the comments, hopefully they are clearer now.

nicoonoclaste reviewed (dismissed):
Looks good, I only have two small suggestions  :+1:

PhDyellow commented:
Thanks for the review and the suggestions @nbraud, I will implement them. Unfortunately, my test machine needed a service, so I won't be able to verify any changes for a week or two. I will push an update when I have the machine back.

PhDyellow commented on pkgs/by-name/ry/ryzen-monitor-ng/package.nix:
```diff
@@ -14,13 +14,14 @@ stdenv.mkDerivation {
 
-  makeTargets = [ "clean" "install" ];
-
   installPhase = ''
```
Is that change worth a separate PR? 

If not, is it worth including now?

JohnRTitor committed:
ryzen-monitor-ng: remove custom installPhase in favor of makeFlags

JohnRTitor committed:
ryzen_monitor_ng: add updateScript, update homepage

apply nixfmt-rfc-style

Signed-off-by: John Titor <50095635+JohnRTitor@users.noreply.github.com>

JohnRTitor committed:
ryzen-monitor-ng: moved cleanup to postfetch

Signed-off-by: John Titor <50095635+JohnRTitor@users.noreply.github.com>

JohnRTitor reviewed (approved):
Hi I have rebased and pushed the suggested fixes. This builds for me, will merge once eval passes

PhDyellow commented:
Apologies, my dev machine still hasn't come back from service. Thanks for pushing this through.
